### PR TITLE
Set hash for identifier transform node to allow merging

### PR DESCRIPTION
**Pull request by kanitw**
Fix https://github.com/vega/vega-lite/issues/4786

### CODE REVIEW

File: src/compile/data/identifier.ts [hunk 1]
```diff
@@ -15,6 +15,10 @@ export class IdentifierNode extends DataFlowNode {
     return new Set([SELECTION_ID]);
   }
 
+  public hash() {
+    return 'Identifier';
```
Comment (domoritz) [inline src/compile/data/identifier.ts]: No `SELECTION_ID`?

Comment (kanitw) [reply src/compile/data/identifier.ts]: `export const SELECTION_ID = '_vgsid_';` --> why would a constant matter?

Comment (domoritz) [reply src/compile/data/identifier.ts]: Right.

Comment (kanitw) [reply src/compile/data/identifier.ts]: Merge?

**Pull request merged**